bundles/org.openhab.binding.icalendar/src/main/java/org/openhab/binding/icalendar/internal/logic/BiweeklyPresentableCalendar.java

```diff
@@ -439,8 +431,47 @@ public Event toEvent() {
             final Summary eventSummary = vEvent.getSummary();
             final String title = eventSummary != null ? eventSummary.getValue() : "-";
             final Description eventDescription = vEvent.getDescription();
-            final String description = eventDescription != null ? eventDescription.getValue() : "";
-            return new Event(title, start, end, description);
+            final String description = eventDescription != null ? eventDescription.getValue() : null;
+            final Location eventLocation = vEvent.getLocation();
+            final String location = eventLocation != null ? eventLocation.getValue() : null;
+            final List<Contact> eventContacts = vEvent.getContacts();
+            final List<String> contactStrings = new ArrayList<>(eventContacts.size());
+            eventContacts.forEach((c) -> contactStrings.add(c.getValue()));
```

Comment:
How about just `final List<String> contactStrings = vEvent.getContacts().stream().map(c -> c.getValue()).toList();`? This way, the local variable `eventContacts` wouldn't even be necessary.

This could be taken even further by joining with the subsequent string concatenation, but I'm not sure if this extension is more readable due to the null handling.
```
final String contact = vEvent.getContacts().stream().map(c -> c.getValue())
    .collect(Collectors.collectingAndThen(Collectors.joining("; "), s -> s.isEmpty() ? null : s));
```

Reply:
Same as for comments.